File: dcp-solutions/2024-11-10/solution.php
```php
function myPow($x, $y) {
    if ($x == 0) return 0;
    
    $result = 1;
    $abs_y = abs($y);
    
    while ($abs_y > 0) {
        if (($abs_y & 1) == 1)
            $result *= $x;
        
        $x *= $x;
        $abs_y >>= 1;
    }
    
    return ($y < 0) ? (1 / $result) : $result;
}
```

File: dcp-solutions/2024-11-10/solution.py
```python
def myPow(x: int, y: int) -> int:
    if x == 0 or y < 0:
        return 0
    
    if y == 1:
        return x
    
    half_pow = myPow(x, y // 2)
    
    result = 1
    
    if y % 2 == 1:
        result *= half_pow * half_pow
    else:
        result *= half_pow
    
    return result
```